File: benchmarks/oopsla25/subjects/145.c
#include <assert.h>
#include <fcntl.h>
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <unistd.h>

void f145(const char* s) {
    int i = 0;
    int f = 0;  // False equivalent in C
    while (i < strlen(s)) {
        f = f || (s[i] == 'a');
        i++;
    }
    assert(f);
}

int main(int argc, char *argv[]) {
  char my_string[10240] = {0};
  if (argc == 1) {
    read(fileno(stdin), my_string, 10240);
  } else {
    int fd = open(argv[1], O_RDONLY);
    read(fd, my_string, 10240);
    close(fd);
  }
  f145(my_string);
  return 0;
}

File: benchmarks/oopsla25/subjects/145.py
import sys

def f145(s: str):
  i = 0
  f = False
  while i < len(s):
    f = f or s[i] == "a"      
    i += 1
  assert f

if __name__ == '__main__':
  my_string = ""
  if len(sys.argv) == 1:
    my_string = sys.stdin.read()
  else:
    fd = open(sys.argv[1], 'r')
    my_string = fd.read()
    fd.close()
  
  f145(my_string)
